pipeline_dp/budget_accounting.py

```diff
@@ -196,11 +247,29 @@ def _normalise_mechanism_weights(self):
         for mechanism in self.mechanisms:
             mechanism.weight *= normalisation_factor
 
+    def _compute_budget_for_aggregation_scope(self):
+        n_aggregations = self.accountant._n_aggregations
+        weights = self.accountant._aggregations_weights
+        if n_aggregations is not None:
+            budget_ratio = 1 / n_aggregations
+        elif weights is not None:
+            sum_weights = sum(weights)
```

Comment:
Are you still able to provide weights when calling `aggregate()`? If so, what happens when that weight is different from the original weight provided to `BudgetAccountant`?

Reply:
yes, now (previously PR wasn't ready yet) they are checked `BudgetAccount.scope()`, that those weights should be the same